packages/features/calendars/DatePicker.tsx

```diff
@@ -138,6 +139,21 @@ const Days = ({
     ? availableDates(props.includedDates)
     : props.includedDates;
 
+  useEffect(() => {
+    if (selected && selected.date()) {
+      if (selected.format("YYYY-MM") === browsingDate.format("YYYY-MM")) return;
+      if (!includedDates) props.onChange(browsingDate.set("date", 1));
```

Comment:
This code repeats the logic that decides which date would be enabled. We should avoid that as it can cause an inconsistency in future.
As the requirement is to select the first enabled date, we can refactor a bit and do exactly that


![image](https://github.com/calcom/cal.com/assets/1780212/631742a2-27a0-43df-af00-b7aad68e4041)

Reply:
Also, while you are fixing this bug, please see if you can also fix the scenario when date query param that is present in the URL is no longer an enabled date e.g. http://org2.cal.local:3000/test/d?date=2023-08-05&month=2023-08
It shows up like this. In this case also, we can select the first enabled date.
![image](https://github.com/calcom/cal.com/assets/1780212/d9984ad1-ca3e-4b3c-98c9-9b6401a83611)

Reply:
We need to have firstEnabledDate of current month. I am not sure why that is happening as in the code I shared above I am finding the firstEnabledDate from days that are to be rendered, so it shouldn't have previous month day.

Reply:
firstEnabledDate looks into the daysToRender. But what if that particular month does not have available dates?. It is completely booked. Then there will be no firstEnabledDate which leads to this misbehaviour by avaliableTimeSlots.